fe/fe-core/src/main/java/org/apache/doris/insertoverwrite/InsertOverwriteManager.java

```diff
@@ -107,44 +107,82 @@ public void registerTaskInGroup(long groupId, long taskId) {
         taskGroups.get(groupId).add(taskId);
     }
 
-    public List<Long> tryReplacePartitionIds(long groupId, List<Long> oldPartitionIds) {
+    /**
+     * this func should in lock scope of getLock(groupId)
+     *
+     * @param newIds if have replaced, replace with new. otherwise itself.
+     * @return
+     */
+    public boolean tryReplacePartitionIds(long groupId, List<Long> oldPartitionIds, List<Long> newIds) {
         Map<Long, Long> relations = partitionPairs.get(groupId);
-        List<Long> newIds = new ArrayList<Long>();
-        for (Long id : oldPartitionIds) {
+        boolean needReplace = false;
+        for (int i = 0; i < oldPartitionIds.size(); i++) {
+            long id = oldPartitionIds.get(i);
             if (relations.containsKey(id)) {
                 // if we replaced it. then return new one.
                 newIds.add(relations.get(id));
             } else {
-                // otherwise itself. we will deal it soon.
-                newIds.add(id);
+                newIds.add(oldPartitionIds.get(i));
```

Comment:
what diffent with `id` and `oldPartitionIds.get(i)`

Reply:
well they're same. will fix.